src/core/task/index.ts

```diff
@@ -1472,9 +1522,8 @@ export class Task {
 		let systemPrompt = await SYSTEM_PROMPT(cwd, supportsBrowserUse, this.mcpHub, this.browserSettings)
 
 		let settingsCustomInstructions = this.customInstructions?.trim()
-		const preferredLanguage = getLanguageKey(
```

Comment:
> Can you explain to me again why you can't do the migrations. I don't want the user's settings to be not honored without them knowing

@frostbournesb The problem was here. Pushing this to the migration code would skip the preferred language to the default of english and this was a finicky behavior(as the process would also interfere by taking the value from the debugger environment it was running inside). I changed this a little bit so now the preferred language would migrate normally.